src/Microsoft.NetCore.Analyzers/Core/Security/UseSecureCookiesASPNetCore.cs

```diff
@@ -0,0 +1,235 @@
+                                                                                        DefinitelyUseSecureCookiesASPNetCoreRule,
+                                                                                        MaybeUseSecureCookiesASPNetCoreRule);
+
+        private static readonly ConstructorMapper constructorMapper = new ConstructorMapper(
+                                                                        ImmutableArray.Create<PropertySetAbstractValueKind>(
+                                                                            PropertySetAbstractValueKind.Flagged));
+
+        private static readonly PropertyMapperCollection PropertyMappers = new PropertyMapperCollection(
+            new PropertyMapper(
+                "Secure",
+                (ValueContentAbstractValue valueContentAbstractValue) =>
+                {
+                    switch (valueContentAbstractValue.NonLiteralState)
+                    {
+                        case ValueContainsNonLiteralState.No:
+                            // We know all values, so we can say Flagged or Unflagged.
+                            return valueContentAbstractValue.LiteralValues.Contains(true)
+                                ? PropertySetAbstractValueKind.Unflagged
+                                : PropertySetAbstractValueKind.Flagged;
+                        case ValueContainsNonLiteralState.Maybe:
+                            // We don't know all values, so we can say Flagged, or who knows.
+                            return valueContentAbstractValue.LiteralValues.Contains(true)
+                                ? PropertySetAbstractValueKind.Unflagged
+                                : PropertySetAbstractValueKind.Unknown;
+                        default:
+                            return PropertySetAbstractValueKind.Unknown;
+                    }
+                }));
+
+        private static HazardousUsageEvaluationResult HazardousUsageCallback(IMethodSymbol methodSymbol, PropertySetAbstractValue propertySetAbstractValue)
+        {
+            switch (propertySetAbstractValue[0])
+            {
+                case PropertySetAbstractValueKind.Flagged:
+                    return HazardousUsageEvaluationResult.Flagged;
+
+                case PropertySetAbstractValueKind.Unflagged:
+                    return HazardousUsageEvaluationResult.Unflagged;
+
+                default:
+                    return HazardousUsageEvaluationResult.MaybeFlagged;
+            }
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            // If there are more classes implement IResponseCookies, add them here later.
```

Comment:
>IResponseCookies [](start = 51, length = 16)

Does PropertySetAnalysis already handle interfaces?  Or am I thinking of TaintedDataAnalysis?  Might be worth testing.

Reply:
I replaced it with WellKnownTypeNames.MicrosoftAspNetCoreHttpResponseCookies then these test cases would fail. For now, it seems PropertySetAnalysis can't support interface.